### PR TITLE
ci: remove duplicate DCO validation

### PR DESCRIPTION
**Pull request by humblec**
This commit remove the duplicate DCO in default rule.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
we are indidually handling the merge rules for devel and release branches, so may be this is not required, however having it in default should also be fine i think. `dependabot` commits are exception, so if we add this in the default rule, it could cause issues. however would like to seek the thoughts here.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
This will block PR's created by dependent bot as  commitlint doesn't passes

File: .mergify.yml
Comment (nixpanic) [inline .mergify.yml]: The commit message does not change when rebasing the PR. Once Mergify decides to put it in this queue, the commitlint check has been validated already (with dependabot commits skipped). There is no need to include commitlint here again (like that, DCO could be removed here too).

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: .mergify.yml
Comment (humblec) [reply .mergify.yml]: yeah , true.. let me just remove DCO for now and the rule conditions can be validated later..

**Review by Madhu-1 — COMMENTED** [review]
update PR description.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
Remove DOC in other `and` conditions too

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
Manual merge is required. merge this once PR's in the queue gets merged https://dashboard.mergify.com/github/ceph/repo/ceph-csi/queues

**Review by nixpanic — APPROVED** [review]

**Pull request merged**
